olp-cpp-sdk-core/include/olp/core/client/OlpClientSettingsFactory.h

```diff
@@ -50,7 +50,7 @@ class CORE_API OlpClientSettingsFactory final {
    * which is defaulted to platform-specific implementation.
    * @return An instance of Network.
    */
-  static std::unique_ptr<http::Network> CreateDefaultNetworkRequestHandler(
+  static std::shared_ptr<http::Network> CreateDefaultNetworkRequestHandler(
```

Comment:
great point. unfortunately some implementations of ```http::Network``` inherit ```std::shared_from_this```. Casting them to base class that knows nothing about weak_ptr inside and then instantiating unique_ptr is UB.

Reply:
That doesn't matter. The CreateDefaultNetworkRequestHandler() static is just to create a platform-dependent instance and pass the ownership to you. You then pass it as shared to OlpClientSettings and from that point it is shared_ptr. How the class itself is designed internally should not be the scope of CreateDefaultNetworkRequestHandler() design.

Reply:
It shouldn't but it is. ```enable_shared_from_this``` requires at least 1 shared_ptr to that object to work properly. Once you have that shared_ptr there is no way back to ```unique_ptr```. If you create ```unique_ptr```, cast it to unique_ptr to base and only then move ownership to ```shared_ptr``` there is no way for weak pointer inside to be initialized because ```shared_ptr``` constructor can't see ```enable_shared_from_this```